django/db/models/base.py

```diff
@@ -1319,13 +1320,23 @@ def _perform_unique_checks(self, unique_checks):
                 if f.primary_key and not self._state.adding:
                     # no need to check for unique primary key when editing
                     continue
-                lookup_kwargs[str(field_name)] = lookup_value
+
+                if getattr(f, "geography", False):
+                    # Cast geography fields to a geometry type.
+                    annotation_label = f"{field_name}_as_geometry"
+                    annotate_kwargs = {
+                        annotation_label: Cast(field_name, f.__class__())
+                    }
+                    qs = qs.annotate(**annotate_kwargs)
+                    lookup_kwargs[annotation_label] = lookup_value
+                else:
+                    lookup_kwargs[str(field_name)] = lookup_value
```

Comment:
Your proposal fixes the issue, however, it's not in the best place, as it mixes `GIS`-logic with `Model`. I'd rather consider fixing the `__exact` lookup for geography fields in all cases, not just when used for uniqueness checking.

Maybe subclassing `PostGISOperator` or using `DatabaseOperations.lookup_cast()` :thinking:

Reply:
Sounds good. In that case, instead of limiting to `__exact`, is it okay to generalize?

Reply:
We know that the `~=` operator is not supported, so `__exact` and `__same_as` are the first candidates:

https://github.com/django/django/blob/5e9aded33f46dd78c5b6050186ba54ff9b02e986/django/contrib/gis/db/backends/postgis/operations.py#L140-L141

If other operators are also not supported, we should also fix them.